tests/system/providers/amazon/aws/example_s3.py

```diff
@@ -202,6 +203,12 @@ def check_fn(files: list) -> bool:
     )
     # [END howto_operator_s3_file_transform]
 
+    # This task skips the `sensor_keys_unchanged` task because the S3KeysUnchangedSensor
```

Comment:
We have similar sensor in google provider for GCS
Was it handled diffrently there?

Reply:
The DebugExecutor runs all sensors in reschedule mode, which is causing problems for us with this sensor, and that executor is used for system tests. I'm not sure how they are working around that, but maybe @bhirsz can weigh in?

Reply:
We overwrite the ``poke`` mode for the sensors (but not for all, I need to investigate the results..):
```
    # [START howto_sensor_object_exists_task]
    gcs_object_exists = GCSObjectExistenceSensor(
        bucket=BUCKET_NAME,
        object=FILE_NAME,
        mode='poke',
        task_id="gcs_object_exists_task",
    )
    # [END howto_sensor_object_exists_task]
```

Other option is to design the tests in a way, that if sensor is in reschedule mode it will not fail on first try (ie. if you're waiting for some file to appear, to upload the file first then run the sensor).

Reply:
> Other option is to design the tests in a way, that if sensor is in reschedule mode it will not fail on first try (ie. if you're waiting for some file to appear, to upload the file first then run the sensor).

Interesting! This would be quite a fundamental change since it's fairly deep in core Airflow code where the sensor mode is hard coded to reschedule, so to hook into that would be tricky.

Can we at least agree to create a new issue to track this possible work so that we can unblock this PR? We're happy to skip this one sensor as this PR suggest, to unblock getting coverage from the rest of this test module.

Reply:
https://github.com/apache/airflow/issues/26725